src/go/rpk/pkg/cli/cluster/partitions/balancer_run.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package partitions
+
+import (
+	"fmt"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newTriggerBalnacerCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "balance",
+		Short: "Triggers on-demand partition balancer",
+		Long: `Triggers on-demand partition balancer.
+
+This command allows you to trigger on-demand partition balancer.
```

Comment:
We probably need a longer _long_ description here. We could possibly explain why the command is needed, requirements (if any), and what's outcome after running this command. 

I would also add this part here:
```
To check the command progress run 'rpk cluster partitions balancer-status'
```

Reply:
Thank you. I will add more details in the long description.

My understanding is this command works independently from continuous balancer. 
@ztlpn, is it safe to run the on-demand balancer any time, i.e. when the continuous balancer is running? If not, probably it's better to hide the command by default.

Reply:
Yes, it is safe.